server/src/test/java/org/elasticsearch/cluster/routing/allocation/BalancedShardsAllocatorTests.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.cluster.routing.allocation;
```

Comment:
Suggest putting this in the same package as the `BalancedShardsAllocator` itself:

```suggestion
package org.elasticsearch.cluster.routing.allocation.allocator;
```

Reply:
Ah sorry this duplicates a previous comment.